### PR TITLE
limit extend.css to html-based transformations based on transtype or out.ext

### PR DESCRIPTION
**Pull request by thendarion**

### CODE REVIEW

**Comment by thendarion** [comment]
at least now I know that if I upload a new commit to the same branch, it updates the pull request too 😃 

anyway, I tried
``` 
  <condition property="is.html">
      <equals arg1="${out.ext}" arg2=".html"/>
  </condition>
```
and it didn't work because there's no out.ext at this point in DITA-OT. transtype-specific settings only come after preprocess it seems 😞

**Comment by jason-fox** [comment]
I've added a [unit-test](https://github.com/jason-fox/fox.jason.extend.css/tree/master/test/non-html) andI think I have this working now. Please reinstall from `master`:

```console
dita uninstall fox.jason.extend.css
dita install https://github.com/jason-fox/fox.jason.extend.css/archive/master.zip
```

And if you can confirm that works I can make a release.

**Comment by thendarion** [comment]
you cannot rely on out.ext because not all DITA-OT html plugins set the property before preprocess and extend.css is part of preprocess, which means that out.ext breaks (I'm looking at you, org.dita.xhtml!) 😞 

I'll ask the DITA-OT people whether that's intentional or a bug 😃

**Comment by thendarion** [comment]
Jarno couldn't remember off top of his head what the out.ext logic is in the plugins so I guess it might get unified at some point. until then, let's make sure that extend.css works in all situations as expected

**Comment by jason-fox** [comment]
So it will just be a matter of getting the `<condition property="is.html">` statement right - see:

https://github.com/jason-fox/fox.jason.extend.css/blob/master/process_css_extend_template.xml#L24


I could add:

```xml
<contains string="${transtype}" substring="HTML" casesensitive="no"/>
<contains string="${transtype}" substring="eclipsehelp" casesensitive="no"/>
```

Does there need to be anything else?

**Comment by thendarion** [comment]
need? no

but I have a couple of ideas :D 

`is.html` is a pretty generic name. I'd be wary of using such names in an ant environment where all properties from all plugins are everywhere

there's no reason to do all the computing of css if you don't copy it at the end, the whole `extend.css` chain should be skipped

`<tempfile>` can `@createfile` so there's no need to `<touch>` separately

`user.csspath` belongs in `extend.css.init` with the other properties logically

`<condition>` can `@else` so there's no need to `<property>` `user.csspath` separately

(I just noticed that there's a comment button in the diff view, so I'll try to add these comments there too)

File: process_css_extend_template.xml [hunk 1]
```diff
@@ -4,38 +4,34 @@
...
         <copy tofile="${output.dir}/${user.csspath}/common-extended.css" file="${extend.css.file}"/>
     </target>
-    
-    <target name="extend.css" depends="extend-css-check-transtype, extend-css-set-properties" if="enable.extend.css">
-        <antcall target="call.extend.css" inheritRefs="true"/>
-    </target>
-    
+
     <target name="extend-css-check-transtype">
         <condition property="enable.extend.css">
```
Comment (thendarion) [inline process_css_extend_template.xml:24]: is.html is a pretty generic name. I'd be wary of using such names in an ant environment where all properties from all plugins are everywhere

File: process_css_extend_template.xml [hunk 2]
```diff
@@ -4,38 +4,34 @@
...
-            <or>
-                <not>
-                    <isset property="args.csspath"/>
-                </not>
-                <isabsolute path="${args.csspath}"/>
-            </or>
-        </condition>
-        <property name="user.csspath" value="${args.csspath}/"/>
+
+    <target name="extend.css" depends="extend-css-check-transtype" if="enable.extend.css">
```
Comment (thendarion) [inline process_css_extend_template.xml:33]: there's no reason to do all the computing of css if you don't copy it at the end, the whole extend.css chain should be skipped

File: process_css_extend_template.xml [hunk 3]
```diff
@@ -4,38 +4,34 @@
...
+                <not>
+                    <isset property="args.csspath"/>
+                </not>
+                <isabsolute path="${args.csspath}"/>
+            </or>
+        </condition>
         <mkdir dir="${dita.temp.dir}"/>
-        <tempfile deleteonexit="true" destdir="${dita.temp.dir}" property="extend.css.file" suffix=".css"/>
-        <touch file="${extend.css.file}"/>
+        <tempfile deleteonexit="true" destdir="${dita.temp.dir}" property="extend.css.file" suffix=".css" createfile="yes"/>
```
Comment (thendarion) [inline process_css_extend_template.xml:16]: `<tempfile>` can `@createfile` so there's no need to `<touch>` separately

File: process_css_extend_template.xml [hunk 4]
```diff
@@ -4,38 +4,34 @@
 --><project xmlns:dita="http://dita-ot.sourceforge.net" name="fox.jason.extend.css">
     
     <target name="extend.css.init">
+        <condition property="user.csspath" value="" else="${args.csspath}/">
```
Comment (thendarion) [inline process_css_extend_template.xml:7]: user.csspath belongs in extend.css.init with the other properties logically

`<condition>` can `@else` so there's no need to `<property>` user.csspath separately

Comment (jason-fox) [reply process_css_extend_template.xml:7]: This shouldn't be the responsibility of this plugin anyway - removed

File: process_css_extend_template.xml (hunk 3 above)
Comment (jason-fox) [reply process_css_extend_template.xml:16]: Added

File: process_css_extend_template.xml (hunk 2 above)
Comment (jason-fox) [reply process_css_extend_template.xml:33]: Calculation shouldn't be in this plugin anyway - removed

File: process_css_extend_template.xml (hunk 1 above)
Comment (jason-fox) [reply process_css_extend_template.xml:24]: Added `<local>` guard

**Comment by jason-fox** [comment]
Fixed in release 1.1.0

**Pull request closed without merging**